main.tf

```diff
@@ -107,34 +89,22 @@ resource "github_branch_protection" "branch_protection_rule" {
   enforce_admins         = local.branch_protection_rules[count.index].enforce_admins
   require_signed_commits = local.branch_protection_rules[count.index].require_signed_commits
 
-  dynamic "required_status_checks" {
-    for_each = local.required_status_checks[count.index]
-
-    content {
-      strict   = required_status_checks.value.strict
-      contexts = required_status_checks.value.contexts
-    }
+  required_status_checks {
+    strict   = lookup(local.branch_protection_rules[count.index].required_status_checks, "strict", null)
+    contexts = lookup(local.branch_protection_rules[count.index].required_status_checks, "contexts", [])
   }
 
-  dynamic "required_pull_request_reviews" {
-    for_each = local.required_pull_request_reviews[count.index]
-
-    content {
-      dismiss_stale_reviews           = required_pull_request_reviews.value.dismiss_stale_reviews
-      dismissal_users                 = required_pull_request_reviews.value.dismissal_users
-      dismissal_teams                 = required_pull_request_reviews.value.dismissal_teams
-      require_code_owner_reviews      = required_pull_request_reviews.value.require_code_owner_reviews
-      required_approving_review_count = required_pull_request_reviews.value.required_approving_review_count
-    }
+  required_pull_request_reviews {
+    dismiss_stale_reviews           = lookup(local.branch_protection_rules[count.index].required_pull_request_reviews, "dismiss_stale_reviews", true)
+    dismissal_users                 = lookup(local.branch_protection_rules[count.index].required_pull_request_reviews, "dismissal_users", [])
+    dismissal_teams                 = lookup(local.branch_protection_rules[count.index].required_pull_request_reviews, "dismissal_teams", [])
+    require_code_owner_reviews      = lookup(local.branch_protection_rules[count.index].required_pull_request_reviews, "require_code_owner_reviews", null)
+    required_approving_review_count = lookup(local.branch_protection_rules[count.index].required_pull_request_reviews, "required_approving_review_count", null)
   }
```

Comment:
i would keep the for_each as we do not want to set this when there are no reviews required..
the reference is just getting crazy complicated.. 
```
for_each = length(keys(local.branch_protection_rules[count.index].required_pull_request_reviews)) > 0 ? .....
```

but maybe also make the resource a `for_each` keyed by branch as we the can just refrence `each.value.required_pull_request_reviews` making it more readable..